pkg/virt-handler/vm.go

```diff
@@ -1365,6 +1367,10 @@ func (d *VirtualMachineController) calculateLiveMigrationCondition(vmi *v1.Virtu
 		return newNonMigratableCondition("VMI uses SEV", v1.VirtualMachineInstanceReasonSEVNotMigratable), isBlockMigration
 	}
 
+	if !topology.AreTSCFrequencyTopologyHintsDefined(vmi) && topology.IsVmiUsingHyperVReenlightenment(vmi) {
```

Comment:
I would adopt the logic you used in `vmi.go`. No need to rework `TopologyHintsForVMI`